osu.Game/Beatmaps/FramedBeatmapClock.cs

```diff
@@ -190,5 +191,28 @@ protected override void Dispose(bool isDisposing)
             base.Dispose(isDisposing);
             beatmapOffsetSubscription?.Dispose();
         }
+
+        public string GetSnapshot()
+        {
+            return
+                $"originalSource: {output(Source)}\n" +
+                $"userGlobalOffsetClock: {output(userGlobalOffsetClock)}\n" +
+                $"platformOffsetClock: {output(platformOffsetClock)}\n" +
+                $"userBeatmapOffsetClock: {output(userBeatmapOffsetClock)}\n" +
+                $"interpolatedTrack: {output(interpolatedTrack)}\n" +
+                $"decoupledTrack: {output(decoupledTrack)}\n" +
+                $"finalClockSource: {output(finalClockSource)}\n";
```

Comment:
I'm not sure this'll be enough either, but it's a start. My follow-up question to this would be, though - how are we going to get this information from users?

If we've hotfixed the issue to not appear (in a clear and jarring manner), and if the seek back is just one frame (which it appears to be from user-provided videos), there's no indication to the player that anything happened, it just may look like a single-frame hitch, wouldn't it? It's not going to be the same freeze until reference clock catches up that the test scene shows, because if it were, users would be getting hard-rewound 12sec back rather than see a single frame 12sec ago?

My point is that if the above is correct then the users won't even _know_ to report this. Are we just going to ask the same people that reported this originally to try playing lots and see if this shows up in their logs? Or maybe there should be an important notification telling the users to go to the issue and attach logs?

Reply:
Yeah, let's auto-report it to sentry.